File: TravelPlannerApp/TravelPlanService/Models/Activity.cs
namespace TravelPlanService.Models
{
    public class Activity
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateTime Date { get; set; }
        public string Time { get; set; } = string.Empty;
        public string Location { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal EstimatedCost { get; set; }
        public string Status { get; set; } = "Planirano";
        public int TravelPlanId { get; set; }
        public TravelPlan TravelPlan { get; set; } = null!;
    }
}

File: frontend/src/models/Activity.js
export const createActivity = (data) => ({
  id:            data.id            || null,
  name:          data.name          || '',
  date:          data.date          || '',
  time:          data.time          || '',
  location:      data.location      || '',
  description:   data.description   || '',
  estimatedCost: data.estimatedCost || 0,
  status:        data.status        || 'Planned',
  travelPlanId:  data.travelPlanId  || null,
});

export const ACTIVITY_STATUSES = ['Planned', 'Reserved', 'Completed', 'Cancelled'];
